Infrastructure/Factories/TribalSpreadsheetTuitionPartnerFactory.cs

```diff
@@ -0,0 +1,542 @@
+                completedLoop = true;
+            }
+            else if (passedHeader)
+            {
+                var data = _spreadsheetExtractor!.GetCellValue(sheetName, dataColumn, row);
+                if (!string.IsNullOrWhiteSpace(data))
+                {
+                    addRowToDictionary(_spreadsheetExtractor, sheetName, data, row);
+                }
+                else
+                {
+                    completedLoop = true;
+                }
+            }
+            else
+            {
+                passedHeader = _spreadsheetExtractor!.GetCellValue(sheetName, tableHeaderColumn, row) == tableHeader;
+            }
+
+            row++;
+        }
+    }
+
+    private TuitionPartner GetOrganisationDetails()
+    {
+        const string KeyColumn = "A";
+        const string ValueColumn = "C";
+        const string TableHeaderColumn = "A";
+        const string TableHeader = "Title";
+
+        TuitionPartner tuitionPartner = new();
+        var sheetName = OrganisationDetailsSheetName;
+
+        foreach (var organisationDetailMapping in _organisationDetailsMapping)
+        {
+            organisationDetailMapping.Value.ClearValue();
+        }
+
+        //Add the data from the spreadsheet in to the _organisationDetailsMapping dictionary value class
+        ProcessSheet(sheetName, TableHeaderColumn, TableHeader, KeyColumn,
+            (spreadsheetExtractor, sheetName, data, row) =>
+            {
+                var key = data;
+                if (_organisationDetailsMapping.ContainsKey(key))
+                {
+                    if (!_organisationDetailsMapping[key].HasConvertedValue)
+                    {
+                        var value = spreadsheetExtractor!.GetCellValue(sheetName, ValueColumn, row);
+                        _organisationDetailsMapping[key].SetValue(value);
+                    }
+                    else
+                    {
+                        _errors.Add($"Duplicate '{key}' in '{sheetName}' worksheet");
+                    }
+                }
+                else
+                {
+                    _warnings.Add($"Unexpected '{key}' key exists in '{sheetName}' worksheet");
+                }
+            });
+
+        //Use the _organisationDetailsMapping dictionary to push the data in to the TuitionPartner class
+        var mappedProperties = _organisationDetailsMapping.Where(x => x.Value.IsStoredInNtp && x.Value.HasConvertedValue);
+        foreach (var mappedProperty in mappedProperties)
+        {
+            mappedProperty.Value.ApplyConvertedValueToProperty(tuitionPartner);
+        }
+
+        //Get Organisation Type Id
+        var organisationTypeName = _organisationDetailsMapping.SingleOrDefault(x => x.Key == "Organisation_LegalStatus_s").Value.SourceValue;
+        var organisationType = _organisationTypes!.Where(x => string.Equals(x.Name, organisationTypeName, StringComparison.InvariantCultureIgnoreCase)).FirstOrDefault();
```

Comment:
It can be simplified with the following: `_organisationTypes!.FirstOrDefault(x => string.Equals(x.Name, organisationTypeName, StringComparison.InvariantCultureIgnoreCase));`

Reply:
Agreed, change made.